storages/backends/azure_storage.py

```diff
@@ -22,6 +22,21 @@
 from storages.utils import to_bytes
 
 
+try:
+    from django_guid import get_guid
+except ImportError:
+    def get_guid():
+        return None
+
+
+def _make_headers():
+    headers = {}
+    correlation_id = get_guid()
+    if correlation_id is not None:
+        headers["x-ms-client-request-id"] = correlation_id
```

Comment:
It seems odd that django-storages doesn't provide a way to set this header somehow. Maybe worth filing an issue or opening a PR. I see [the Pulp team has had some success with extending AzureStorage's `url` method to add params](https://github.com/jschneier/django-storages/pull/1071).

Reply:
I wasn't proposing that we use that. Only that Pulp has had some success in contributing a somewhat similar feature in the past to django-storages.